hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoSqlConnectorBase.java

```diff
@@ -0,0 +1,229 @@
+import com.hazelcast.jet.sql.impl.connector.SqlProcessors;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.row.JetSqlRow;
+import com.hazelcast.sql.impl.schema.ConstantTableStatistics;
+import com.hazelcast.sql.impl.schema.MappingField;
+import com.hazelcast.sql.impl.schema.Table;
+import com.hazelcast.sql.impl.schema.TableField;
+import org.apache.calcite.rex.RexNode;
+import org.bson.BsonDocument;
+import org.bson.BsonTimestamp;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.hazelcast.jet.core.Edge.between;
+import static com.hazelcast.jet.mongodb.impl.Mappers.bsonDocumentToDocument;
+import static com.hazelcast.jet.mongodb.impl.Mappers.defaultCodecRegistry;
+import static com.hazelcast.sql.impl.type.QueryDataType.OBJECT;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Base for MongoDB SQL Connectors.
+ * <p>
+ * Streaming and batch connectors have similar way of dealing with scans with a few exceptions (like the requirement
+ * for {@code _id field}).
+ * <p>
+ * All MongoDB connectors assume at least one primary key.
+ * If user didn't specify any, the {@code _id} column is set to primary key - it is mandatory (auto-created if not specified
+ * by user), unique and indexed.
+ * <p>
+ *
+ * @see FieldResolver
+ */
+public abstract class MongoSqlConnectorBase implements SqlConnector {
+
+    @Nonnull
+    @Override
+    public List<MappingField> resolveAndValidateFields(
+            @Nonnull NodeEngine nodeEngine,
+            @Nonnull Map<String, String> options,
+            @Nonnull List<MappingField> userFields,
+            @Nonnull String externalName
+    ) {
+        FieldResolver fieldResolver = new FieldResolver(nodeEngine);
+        return fieldResolver.resolveFields(externalName, options, userFields, isStream());
+    }
+
+    @Nonnull
+    @Override
+    public List<String> getPrimaryKey(Table table) {
+        MongoTable mongoTable = (MongoTable) table;
+        return singletonList(mongoTable.primaryKeyName());
+    }
+
+    @Nonnull
+    @Override
+    public Table createTable(@Nonnull NodeEngine nodeEngine, @Nonnull String schemaName, @Nonnull String mappingName,
+                             @Nonnull String collectionName, @Nonnull Map<String, String> options,
+                             @Nonnull List<MappingField> resolvedFields) {
+        FieldResolver fieldResolver = new FieldResolver(nodeEngine);
+        String databaseName = Options.getDatabaseName(nodeEngine, options);
+        String pkName = Options.getPkColumn(options, isStream());
```

Comment:
isStream() can aassigned to a variable and used, instead of calling isStream() multiple times. Also it iseems pkName is not used